app/routes/questionnaire.py

```diff
@@ -179,33 +183,46 @@ def get_section(schema, questionnaire_store, section_id, list_item_id=None):
         questionnaire_store.metadata,
     )
 
-    if not schema.is_hub_enabled():
-        redirect_location = router.get_first_incomplete_location_in_survey()
-        return redirect(redirect_location.url())
-
     if section_id not in router.enabled_section_ids:
         return redirect(url_for(".get_questionnaire"))
 
-    routing_path = router.routing_path(section_id=section_id, list_item_id=list_item_id)
-    section_status = questionnaire_store.progress_store.get_section_status(
-        section_id=section_id, list_item_id=list_item_id
-    )
+    if request.method == "GET":
+        section_status = questionnaire_store.progress_store.get_section_status(
+            section_id=section_id, list_item_id=list_item_id
+        )
 
-    if section_status == CompletionStatus.COMPLETED:
-        return redirect(
-            router.get_section_return_location_when_section_complete(routing_path).url()
+        routing_path = router.routing_path(
+            section_id=section_id, list_item_id=list_item_id
         )
 
-    if section_status == CompletionStatus.NOT_STARTED:
+        if section_status != CompletionStatus.COMPLETED:
+            return redirect(
+                router.get_first_incomplete_location_for_section(routing_path).url()
+            )
+
+        if schema.is_summary_in_section(section_id):
+            section_summary = _get_section_summary(
+                list_item_id, questionnaire_store, schema, section_id
+            )
+            return _render_page(
+                block_type="SectionSummary",
+                context=section_summary.context(),
+                current_location=section_summary.current_location,
+                previous_location_url=section_summary.get_previous_location_url(),
+                schema=schema,
+                page_title=section_summary.page_title,
+            )
+
         return redirect(
-            router.get_first_incomplete_location_for_section(routing_path).url()
+            router.get_first_incomplete_location_for_section(
+                routing_path=routing_path
+            ).url()
         )
 
-    return redirect(
-        router.get_first_incomplete_location_for_section(
-            routing_path=routing_path
-        ).url()
+    section_summary = _get_section_summary(
+        list_item_id, questionnaire_store, schema, section_id
     )
+    return redirect(section_summary.get_next_location_url())
```

Comment:
There is a lot going on here and I think more should be handled by the handler. Ideally the handler should abstract the behaviour away from the route so we can highlight the places where redirects happen. This code won't work as methods would need to be implemented in the handler, but this could look something like:

```python
    try:
        section_handler = SectionHandler(
            schema=schema,
            questionnaire_store=questionnaire_store,
            section_id=section_id,
            list_item_id=list_item_id,
            language=flask_babel.get_locale().language
        )
    except InvalidLocationException:
        return redirect(url_for(".get_questionnaire"))

    if request.method == "GET":
        if section_handler.can_display_summary():
            return _render_page(
                block_type="SectionSummary",
                context=section_handler.context(),
                current_location=section_handler.current_location,
                previous_location_url=section_handler.get_previous_location_url(),
                schema=schema,
                page_title=section_handler.page_title,
            )

        return redirect(section_handler.get_section_resume_url())

    return redirect(section_handler.get_next_location_url())
```
- The `InvalidLocationException` would deal with the `router.enabled_section_ids` check
- `section_handler.can_display_summary` would deal with checking that a section summary exists and the section is complete
- `get_section_resume_url` would deal with where the user should resume i.e. the first location if the section is complete or the first incomplete location